crates/revm/src/db/in_memory_db.rs

```diff
@@ -41,6 +33,12 @@ pub struct CacheDB<ExtDB: DatabaseRef> {
     pub db: ExtDB,
 }
 
+impl<ExtDB: DatabaseRef + Default> Default for CacheDB<ExtDB> {
```

Comment:
Not really liking this, having default is okay when you don't have any inputs. but this would require specifying `<DB>` generic item, so it is better to go simple and just have `fn new`

Reply:
The generic has to be specified when doing `CacheDB::new(Default::default())` too, which now is the same as `CacheDB::<DB>::default()`

Reply:
My point exactly, but with usage I would usually type `CacheDB::new(` and add check what rust-analyzer is giving me as use `CacheDB::new(DB::default())`, having generic there complicates this workflow and to the point, it is the same outcome but more complex. Simple and stupid is in most times a better way.

Reply:
Right, which is why `fn new` is unchanged for if you want the basic empty DB. Implementing Default allows having `CacheDB<MyDB>: Default` where `MyDB: Default`, which also allows deriving `Default` on structs containg it etc. I don't see anything wrong with this

Reply:
Hm, not sure why i thought this was exclusive as one would disable the other. It is good to add it.

Just check `ExtDB: DatabaseRef + Default` it should be `ExtDB: Database + Default` as `Database` is the main used trait.

Reply:
`CacheDB` is defined with `ExtDB: DatabaseRef`, maybe in another PR?

Reply:
I see, strange tbh. Yeah, another time, it will break things if we change it.